### PR TITLE
Making 'AliasStore' suitable for holding things other than Strings

### PR DESCRIPTION
**Pull request by lewisd32**
Makes `AliasStore` generic so it can hold things other than Strings.
Includes a test demonstrating this with `UUID`.
Also extracts the literal, empty, and ABSENT handling into wrappers so they can be composed.

### CODE REVIEW

File: unacceptable-aliases/src/main/java/io/github/unacceptable/alias/PasswordGenerator.java [hunk 1]
```diff
@@ -43,6 +44,12 @@ public PasswordGenerator(String passwordChars) {
     public PasswordGenerator(int length, String passwordChars) {
         this.length = length;
         this.passwordChars = passwordChars;
+        generator = new EmptyWrappingGenerator(new AbsentWrappingGenerator(new LiteralWrappingGenerator(this::generatePassword)));
```
Comment (ojacobson) [inline unacceptable-aliases/src/main/java/io/github/unacceptable/alias/PasswordGenerator.java]: 1. `this.generator` would make this read more smoothly, I think.

2. If we're going to repeat this stack of generators inside of each constructor, we should probably make a factory for it somewhere, or they're going to diverge over time, I think.

**Comment by ojacobson** [comment]
I broadly like this otherwise.

**Comment by lewisd32** [comment]
I'm happy with this, it's ready for a re-review and merge when you're happy.

File: unacceptable-aliases/src/test/java/io/github/unacceptable/alias/PasswordGeneratorTest.java [hunk 2]
```diff
@@ -4,8 +4,7 @@
 
 import static io.github.unacceptable.matchers.PatternMatcher.pattern;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.*;
```
Comment (ojacobson) [inline unacceptable-aliases/src/test/java/io/github/unacceptable/alias/PasswordGeneratorTest.java]: Thanks, IDEA's default settings!

(Don't worry about removing this from the PR. I'll hit it next time I'm near this test.)

**Comment by ojacobson** [comment]
Released in v0.1.0.

File: unacceptable-aliases/src/test/java/io/github/unacceptable/alias/PasswordGeneratorTest.java (hunk 2 above)
Comment (lewisd32) [inline unacceptable-aliases/src/test/java/io/github/unacceptable/alias/PasswordGeneratorTest.java]: Doh! I set that correctly on one machine, but I guess not on the other. Or maybe I lost it when I reimported.

**Pull request merged**